Jenkinsfile

```diff
@@ -150,7 +150,7 @@ pipeline {
                     environment {
                         GITHUB_SSH_KEY = credentials('github_bot_ssh_key')
                         PATH = "/home/ubuntu/bin/:${env.PATH}"
-                        KONG_SOURCE = "3.0.0.0"
+                        KONG_SOURCE = "master"
```

Comment:
why was this done? what purpose does this serve? did you understand the intent previously before you changed this or did you just change this to "make it work?"

We now have two stages that run against Kong EE master and none that run against 3.0.0.0 so we no longer have a backwards compatibility CI check. If we no longer care about backwards compatibility why not remove the entire check instead of run kong-ee@master twice?

Reply:
The "backward compat" is testing a behaviour introduced by a bug,
kbt is not pinned correctly https://github.com/Kong/kong-ee/pull/4238/files and always pointing to master. So I don't
understand what are we testing here to use the master branch of this
repo against 3.0.0.0; unless we intended to always build 3.0.0.0 with master kbt.
I saw a commit and PR (https://github.com/Kong/kong-build-tools/pull/622)
that doesn't tell me anything why it's being added, without knowing why 
it was there, i'm inclined to just remove it and instead make it pass. That being said, i'm okay to just remove this.

Patches will stay in kong/kong-ee repo in the future, hopefully at that time we don't need to try 5 different places to correctly bump/pin a dependency.